picard/similarity.py

```diff
@@ -53,12 +53,22 @@ def similarity(a1, b1):
 
 def similarity2(a, b):
     """Calculates similarity of a multi-word strings."""
+    if not a or not b:
+        return 0.0
+    if a == b:
+        return 1.0
```

Comment:
I think the equality check should go first. I see tests currently go the opposite, but I'd expect `similarity2("", "") == 1.0`. What did the old code do?

Reply:
I was also expecting `similarity2("", "") == 1.0` but that's not what the old code does, so I preserved this behavior in order to ensure nothing breaks.